Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -542,6 +540,52 @@ - (NSInteger)getBranchViewCount:(NSString *)branchViewID {
     return count;
 }
 
+- (void)saveBranchAnalyticsData:(NSDictionary *)analyticsData {
+    if (_sessionID) {
+        if (!_savedAnalyticsData) {
+            _savedAnalyticsData = [self getBranchAnalyticsData];
+        }
+        NSMutableArray *viewDataArray = [_savedAnalyticsData objectForKey:_sessionID];
+        if (!viewDataArray) {
+            viewDataArray = [[NSMutableArray alloc] init];
+            [_savedAnalyticsData setObject:viewDataArray forKey:_sessionID];
+        }
+        [viewDataArray addObject:analyticsData];
+        [self writeObjectToDefaults:KEY_BRANCH_ANALYTICAL_DATA value:_savedAnalyticsData];
+    }
+}
+
+- (void)clearBranchAnalyticsData {
+    [self writeObjectToDefaults:KEY_BRANCH_ANALYTICAL_DATA value:nil];
+    _savedAnalyticsData = nil;
+}
+
+- (NSMutableDictionary *)getBranchAnalyticsData {
+    NSMutableDictionary *analyticsDataObj = _savedAnalyticsData;
+    if (!analyticsDataObj) {
+        NSObject *dataObj = [self readObjectFromDefaults:KEY_BRANCH_ANALYTICAL_DATA];
+        if (dataObj) {
+            analyticsDataObj = (NSMutableDictionary *)dataObj;
+        } else {
+            analyticsDataObj = [[NSMutableDictionary alloc] init];
+        }
```

Comment:
Good catch!
Was scared of any potential issue with  typecasting a  `nil`. Just learned it is not an issue on iOS